XcodeServerSDK/XcodeServerConfig.swift

```diff
@@ -33,16 +42,51 @@ public class XcodeServerConfig : JSONSerializable {
         return dict
     }
     
-    public init(var host: String, user: String?, password: String?) {
+    ///
+    /// Initializes a server configuration with the provided host.
+    /// - parameter host: `Xcode` server host.
+    /// - paramater user: Username that will be used to authenticate against the `host` provided.
+    /// Can be `nil`.
+    ///
+    /// - parameter password: Password that will be used to authenticate against the `host` provided.
+    /// Can be `nil`.
+    ///
+    /// - returns: A fully initialized `XcodeServerConfig` instance.
+    ///
+    /// - throws:
+    ///     - `InvalidHostProvided`: When the host provided doesn't produce a valid `URL`
+    ///     - `InvalidSchemeProvided`: When the provided scheme is not `HTTPS`
+    ///
+    public required init(var host: String, user: String?, password: String?) throws {
+        guard let url = NSURL(string: host) else {
+            /*******************************************************************
```

Comment:
«if» they fix it.

I'll give it a try; just pushed a code that fixes the tests that were complaining due to the lack of `try`.

Also; I think @cojoj mentioned something about how to test throwing functions (but couldn't find it for the live of me on the project)

So here's a test that now passes when the init throws

```swift
    func testInvalidDictionaryInit() {
        let json = [
            "user": "ICanCreateBots",
            "password": "superSecr3t"
        ]
        
        do {
            _ = try XcodeServerConfig(json: json)
        } catch ConfigurationErrors.NoHostProvided {
            print("• `ConfigurationErrors.NoHostProvided` was thrown as expected.")
        } catch {
            XCTFail("`XcodeServerConfig` init failed with an unexpected error: \(error).")
        }
    }
```

I'm trying the init; catching the specific error I'm expecting and logging it to the console; else I'm failing; would be good to fail below the try line now that I think about it.